src/Functions/toRelativeYearNum.cpp

```diff
@@ -11,7 +11,41 @@ using FunctionToRelativeYearNum = FunctionDateOrDateTimeToSomething<DataTypeUInt
 
 REGISTER_FUNCTION(ToRelativeYearNum)
 {
-    factory.registerFunction<FunctionToRelativeYearNum>();
+    FunctionDocumentation::Description description_toRelativeYearNum = R"(
+Converts a date or date with time to the number of years elapsed since a certain fixed point in the past.
+    )";
+    FunctionDocumentation::Syntax syntax_toRelativeYearNum = R"(
+toRelativeYearNum(date)
+    )";
+    FunctionDocumentation::Arguments arguments_toRelativeYearNum = {
+        {"date", "Date or date with time. [`Date`](../data-types/date.md)/[`DateTime`](../data-types/datetime.md)/[`DateTime64`](../data-types/datetime64.md)."}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_toRelativeYearNum = "Returns the number of years from a fixed reference point in the past. [`UInt16`](../data-types/int-uint.md).";
+    FunctionDocumentation::Examples examples_toRelativeYearNum = {
+        {"Get relative year numbers", R"(
+SELECT
+    toRelativeYearNum(toDate('2002-12-08')) AS y1,
```

Comment:
Re my above^^ comment, that is actually standalone usage and not recommended.
Something like

```sql
SELECT
    toRelativeYearNum(toDate('2002-12-08')) - toRelativeYearNum(toDate('2010-10-26'))
```

would be a better example (more idiomatic).